client/conn.go

```diff
@@ -357,6 +363,20 @@ func (c *Conn) SetCharset(charset string) error {
 	}
 }
 
+func (c *Conn) SetCollation(collation string) error {
+	if c.status == 0 {
```

Comment:
I'm a bit worried that after handshake `status` can also be `0` (will check it later). It's safer to use other fields like `c.serverVersion`?